### ISSUE TITLE
Infinite loop on circular bookmarks

### ISSUE THREAD

**Issue by mara004** [pymupdf/PyMuPDF#1505]
## Describe the bug (mandatory)
Calling the method `fitz.Document.get_toc()` on a PDF with circular bookmark references results in an infinite loop.
The Python process needs to be terminated by force using the task manager.

## To Reproduce (mandatory)
```python3
import fitz as pymupdf

mu_doc = pymupdf.Document("./pdfium_circularity.pdf")

for _levelno, title, _pageno, dest in mu_doc.get_toc(simple=False):
    
    page  = _pageno  - 1
    level = _levelno - 1
    mu_page  = mu_doc[page]
    
    is_closed = False
    left = None
    top = None
    zoom = None
    
    if 'collapse' in dest:
        is_closed = dest['collapse']
    if 'to' in dest:
        target_pos = dest['to']
        left = round(mu_page.mediabox[0] + target_pos[0], 3)
        top  = round(mu_page.mediabox[3] - target_pos[1], 3)
    if 'zoom' in dest:
        zoom = dest['zoom']
    
    if is_closed:
        state = '-'
    else:
        state = '+'
    
    print("   "*level + f"[{state}] {title} -> {page}  # {left} {top} {zoom}")
```
[infiniteloop.py.txt](https://github.com/pymupdf/PyMuPDF/files/7797125/infiniteloop.py.txt)
[pdfium_circularity.pdf.txt](https://github.com/pymupdf/PyMuPDF/files/7797126/pdfium_circularity.pdf.txt) (https://pdfium.googlesource.com/pdfium/+/bdb330ee4d4d9d6fafc6bc84d4fa28c045eda936/testing/resources/bookmarks_circular.pdf)

## Expected behavior (optional)
No infinite loop should happen.

## Your configuration (mandatory)
 - Linux KDE Neon User (64-bit)
 - Python 3.8.10
 - PyMuPDF 1.19.3, installation method: wheel

**Comment by JorjMcKie** [comment]
Thanks for the well-prepared report!
Found and fixed the error - which only happens with `get_toc(False)`, in a subroutine that assigns xref numbers to the outline items.

**Comment by JorjMcKie** [comment]
Fixed in v1.19.4 currently being uploaded.
